### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by MeqdamAlqudah**
- Create a class Classroom with the following:
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).
- Create a class Book
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. 
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+    @students = []
+  end
+
+  def add_student(student)
+    student.classroom = @label
+    @students.push(student)
+  end
+end
+
 class Student < Person
```
Comment (er-contreras) [inline student.rb:18]: - [ ] When setting the classroom for a student it also adds it to the classrooms' students:

Check for a method that serves as the 'belongs-to' side (a student belongs to a classroom)
You could do something like this:

```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: school_library_people.rb [hunk 2]
```diff
@@ -36,8 +41,44 @@ def is_of_age?
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (er-contreras) [inline school_library_people.rb]: - [ ] The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: school_library_people.rb [hunk 3]
```diff
@@ -8,14 +8,19 @@ def correct_name(name)
...
     @age = age
     @parent_permission = parent_permission
     @id = Random.rand(1..1000)
     @corrector = Corrector.new
+    @rentals = []
+  end
+
+  def add_rental(_rental)
+    @rentals.push(rentals)
   end
```
Comment (er-contreras) [inline school_library_people.rb]: - [ ] The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @MeqdamAlqudah, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️

- [ ] It would be great if you can separate each Class in his own file or model to keep it cleaner. 😉
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @MeqdamAlqudah 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**